intTests/test_mir_verify_slices/test.saw

```diff
@@ -0,0 +1,115 @@
+enable_experimental;
+
+let oneThroughFive = mir_term {{ [1, 2, 3, 4, 5] : [5][32] }};
+
+let f_spec_1 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_value a];
+
+  mir_return (mir_term {{ 3 : [32] }});
+};
+
+let f_spec_2 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 1 3];
+
+  mir_return (mir_term {{ 5 : [32] }});
+};
+
+// mir_slice_value must take an array reference as an argument.
+// Passing a bare array constitutes a type error.
+let f_fail_spec_1 = do {
+  let arr = mir_array_value mir_u32 [mir_term {{ 1 : [32] }}];
+  mir_execute_func [mir_slice_value arr];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
+
+// The end value of the range given to mir_slice_range_value must be less
+// than the length of the slice.
+let f_fail_spec_2 = do {
+  a <- mir_alloc (mir_array 5 mir_u32);
+  mir_points_to a oneThroughFive;
+
+  mir_execute_func [mir_slice_range_value a 0 5];
+
+  mir_return (mir_term {{ 0 : [32] }});
+};
```

Comment:
This test will need edits after changing to half-open ranges.

Reply:
Indeed, this test case relied on my earlier typo. I've tweaked the test case to use a slice over the range `[0..6]` so that the end now exceeds the length of the array, resulting in the error I was intending to test for.